### PR TITLE
Compile error cfg flag in DispatcherIO when missing tokio and compio

### PR DESCRIPTION
**Pull request by joseph-isaacs**

### CODE REVIEW

File: vortex-io/src/dispatcher/mod.rs [hunk 1]
```diff
@@ -75,7 +73,7 @@ impl Default for IoDispatcher {
         #[cfg(all(feature = "compio", not(feature = "tokio")))]
         return Self(Inner::Compio(CompioDispatcher::new(1)));
         #[cfg(not(any(feature = "compio", feature = "tokio")))]
-        vortex_panic!("must enable one of compio or tokio to use IoDispatcher");
+        compile_error!("must enable one of compio or tokio to use IoDispatcher");
```
Comment (joseph-isaacs) [inline vortex-io/src/dispatcher/mod.rs]: I would think we want this top level?

File: vortex-io/src/dispatcher/mod.rs [hunk 2]
```diff
@@ -5,15 +5,16 @@ mod tokio;
...
-use vortex_error::vortex_panic;
 use vortex_error::VortexResult;
 
 #[cfg(feature = "compio")]
 use self::compio::*;
 #[cfg(feature = "tokio")]
 use self::tokio::*;
 
+#[cfg(not(any(feature = "compio", feature = "tokio")))]
+compile_error!("must enable one of compio or tokio to use IoDispatcher");
```
Comment (AdamGS) [inline vortex-io/src/dispatcher/mod.rs:16]: LGTM, long term we might want to have a default feature here.

**Review by AdamGS — APPROVED** [review]

**Comment by AdamGS** [comment]
actually, the things are set up right now this will just block us from testing other code with default features.

**Comment by joseph-isaacs** [comment]
I updated CI.

I don't fully follow your comment. I think CI should either that this no defaults is valid or that its a comp error

**Comment by danking** [comment]
There’s an error in the CI logs but it looks like that didn’t fail the check / step.

should the entire IO dispatcher stuff get disabled if no asynchronous runtime is available? Or does this package only make sense with some runtime? If the latter, I suppose cargo makes it annoying to disable the default runtime, were to make one of them default?

**Comment by joseph-isaacs** [comment]
> There’s an error in the CI logs but it looks like that didn’t fail the check / step.

This is the point of the change. It goes seem like maybe a default is better?

**Comment by a10y** [comment]
I think this has been supplanted by #1653

**Pull request closed without merging**